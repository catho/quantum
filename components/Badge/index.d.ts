import { FC, ReactNode } from 'react';

export interface BadgeProps {
    skin?: 'primary' | 'secondary' | 'success' | 'error' | 'neutral';
    children?: ReactNode[] | ReactNode;
    number?: number;
    inverted?: boolean;
    dot?: boolean;
}

declare const Badge: FC<BadgeProps>;
export default Badge;
