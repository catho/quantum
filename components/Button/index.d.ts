import React from 'react';
import IconButton from './IconButton';

export interface ButtonProps {
    center?: boolean;
    disabled?: boolean;
    stroked?: boolean;
    full?: boolean;
    icon?: string;
    size?: 'xsmall' | 'small' | 'medium' | 'large' | 'xlarge';
    skin?: 'neutral' | 'primary' | 'secondary' | 'success' | 'warning' | 'error';
    type?: 'button' | 'reset' | 'submit';
    children?: React.ReactNode[] | React.ReactNode;
    $as?: JSX.Element | string;
    onClick?: React.MouseEventHandler<HTMLButtonElement>;
    theme?: {
        baseFontSize?: number;
        colors?: object;
        spacing?: object;
        breakpoints?: object;
        components?: {
            button: object;
        };
    };
    id?: string;
}

export default class Button extends React.Component<ButtonProps> {
    static Icon: IconButton;
}
