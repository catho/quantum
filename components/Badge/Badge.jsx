import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import {
  components,
  spacing,
  baseFontSize as defaultBaseFontSize,
} from '../shared/theme';

const getColors = ({
  skin,
  theme: {
    components: {
      badge: {
        skins: { [skin]: badgeColor },
      },
    },
  },
  inverted,
}) => {
  if (inverted) {
    const [text, background] = [badgeColor.background, badgeColor.text];
    return css`
      background-color: ${background};
      color: ${text};
    `;
  }
  return css`
    background-color: ${badgeColor.background};
    color: ${badgeColor.text};
  `;
};

const BadgeWrapper = styled.div`
  display: inline-block;

  ${props =>
    props.originalChildren
      ? `position: relative;`
      : `margin-left: 8px; margin-right: 8px;`}
`;

const StyledBadge = styled.span`
  border-radius: 8px;
  box-sizing: border-box;
  display: flex;
  font-weight: bold;
  height: 24px;
  line-height: 24px;
  min-width: 32px;
  align-items: center;
  justify-content: center;

  ${({
    value,
    theme: {
      spacing: { xxxsmall, xxsmall },
      baseFontSize,
    },
    number,
  }) => {
    const padding =
      !Number.isInteger(value) || number >= 10 ? xxsmall : xxxsmall;

    return `
        font-size: ${baseFontSize * 0.75}px;
        padding-left: ${padding}px;
        padding-right: ${padding}px;
      `;
  }}

  ${props =>
    props.originalChildren &&
    `
    position: absolute;
    right: -10px;
    top: -10px;
  `}

  ${props =>
    props.dot &&
    `
    right: -2px;
    top: 0;
    height: 12px;
    width: 12px;
    min-width: 12px;
  `}

  ${getColors}
`;

/** This components is used to display only `Numbers`. If you want to pass a string, use `<Tag />` component instead */
const Badge = ({ children, number, skin, inverted, dot, theme }) => {
  const value = number > 99 ? '99+' : number;

  return (
    <BadgeWrapper value={value} originalChildren={children}>
      <StyledBadge
        skin={skin}
        inverted={inverted}
        theme={theme}
        value={value}
        originalChildren={children}
        dot={dot}
      >
        {dot || value}
      </StyledBadge>
      {children}
    </BadgeWrapper>
  );
};

BadgeWrapper.displayName = 'BadgeWrapper';
StyledBadge.displayName = 'StyledBadge';
Badge.displayName = 'Badge';

Badge.propTypes = {
  /** Define background and text color */
  skin: PropTypes.oneOf([
    'primary',
    'secondary',
    'success',
    'error',
    'neutral',
  ]),
  /** When passed a children to <Badge />, the badge will be displayed at top-right corner of the children. */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  /** Number to be displayed inside badge. When number is higher than 99 will be displayed "99+" instead of number value */
  number: PropTypes.number,
  /** Swap background and text color */
  inverted: PropTypes.bool,
  dot: PropTypes.bool,
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
    spacing: PropTypes.object,
    components: PropTypes.shape({
      badge: PropTypes.object,
    }),
  }),
};

Badge.defaultProps = {
  skin: 'neutral',
  inverted: false,
  children: '',
  number: 0,
  dot: false,
  theme: {
    baseFontSize: defaultBaseFontSize,
    spacing,
    components: {
      badge: components.badge,
    },
  },
};

export default Badge;
