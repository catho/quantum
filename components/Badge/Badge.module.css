.wrapper {
    display: inline-block;
    margin-left: var(--qtm-spacing-xsmall);
    margin-right: var(--qtm-spacing-xsmall);
}

.wrapper-content {
    position: relative;
    margin-left: initial;
    margin-right: initial;
}

.badge {
    border-radius: 8px;
    box-sizing: border-box;
    display: flex;
    font-weight: bold;
    height: 24px;
    line-height: 24px;
    min-width: 32px;
    align-items: center;
    justify-content: center;
    font-size: calc(var(--qtm-base-font-size) * 0.75);
    padding-left: var(--qtm-spacing-xxxsmall);
    padding-right: var(--qtm-spacing-xxxsmall);
}

.badge-xxsmall {
    padding-left: var(--qtm-spacing-xxsmall);
    padding-right: var(--qtm-spacing-xxsmall);
}

.badge-content {
    position: absolute;
    right: -10px;
    top: -10px;
}

.badge-dot {
    right: -2px;
    top: 0;
    height: 12px;
    width: 12px;
    min-width: 12px;
}

.badge-skin-neutral {
    background-color: var(--qtm-colors-neutral-100);
    color: var(--qtm-colors-neutral-700);
}

.badge-skin-primary {
    background-color: var(--qtm-colors-primary-700);
    color: var(--qtm-colors-neutral-100);
}

.badge-skin-secondary {
    background-color: var(--qtm-colors-secondary-500);
    color: var(--qtm-colors-neutral-0);
}

.badge-skin-error {
    background-color: var(--qtm-colors-error-500);
    color: var(--qtm-colors-neutral-100);
}

.badge-skin-success {
    background-color: var(--qtm-colors-success-100);
    color: var(--qtm-colors-success-900);
}

.badge-skin-neutral-inverted {
    background-color: var(--qtm-colors-neutral-700);
    color: var(--qtm-colors-neutral-100);
}

.badge-skin-primary-inverted {
    background-color: var(--qtm-colors-neutral-100);
    color: var(--qtm-colors-primary-700);
}

.badge-skin-secondary-inverted {
    background-color: var(--qtm-colors-neutral-0);
    color: var(--qtm-colors-secondary-500);
}

.badge-skin-error-inverted {
    background-color: var(--qtm-colors-neutral-100);
    color: var(--qtm-colors-error-500);
}

.badge-skin-success-inverted {
    background-color: var(--qtm-colors-success-900);
    color: var(--qtm-colors-success-100);
}