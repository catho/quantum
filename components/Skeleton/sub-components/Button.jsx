import classNames from 'classnames';
import PropTypes from 'prop-types';
import Rect from './Rect';
import styles from './Button.module.css';

const Button = ({ size = 'medium', className, ...rest }) => {
  const buttonClass = classNames(
    styles.button,
    styles[`button-${size}`],
    className,
  );

  return <Rect {...rest} className={buttonClass} />;
};

Button.propTypes = {
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
};

export default Button;
