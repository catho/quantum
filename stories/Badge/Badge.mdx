import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';

import Badge from '../../components/Badge';

import { Header } from '../shared';

import { 
  Default,
  WithChildren,
  WithSkin,
  CustomNumber,
  Dot,
  Inverted
} from './Badge.stories.jsx';

<Meta title="Badge" component={Badge} />

<Header title="Badge">
  Badges are used for showing the quantity of something, such as a warning, inbox messages and others.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import Badge from '@catho/quantum/Badge';
```

<Canvas>
  <Story of={Default} />
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Badge} />

## <a id="examples"></a>Examples 

### With children

<Canvas>
  <Story of={WithChildren} />
</Canvas>

### With skin

<Canvas>
  <Story of={WithSkin} />
</Canvas>

### Custom number

<Canvas>
  <Story of={CustomNumber} />
</Canvas>

### Dot

<Canvas>
  <Story of={Dot} />
</Canvas>

### Inverted

<Canvas>
  <Story of={Inverted} />
</Canvas>

