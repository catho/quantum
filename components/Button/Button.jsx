import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { shadow, hexToRgba, theme as defaultTheme } from '../shared';
import { query } from '../Grid/sub-components/shared';

import {
  components,
  spacing,
  colors as defaultColors,
  baseFontSize as defaultBaseFontSize,
} from '../shared/theme';

import Icon from '../Icon/Icon';

const buttonIconBreakpoints = (props, breakpoint) => {
  const q = query(props.theme.breakpoints)[breakpoint];

  return q`
    margin-right: ${props.theme.spacing[breakpoint] / 2}px;
  `;
};

const ButtonIcon = styled(Icon)`
  pointer-events: none;

  ${props =>
    Object.keys(props.theme.breakpoints).map(breakpoint =>
      buttonIconBreakpoints(props, breakpoint),
    )}
`;

const buttonFontAndLineProps = ({ size, theme: { baseFontSize } }) => {
  const sizes = {
    xsmall: `${baseFontSize * 0.75}px`,
    small: `${baseFontSize * 0.75}px`,
    medium: `${baseFontSize}px`,
    large: `${baseFontSize * 1.25}px`,
    xlarge: `${baseFontSize * 1.5}px`,
  };
  const lineHeights = {
    xsmall: `${baseFontSize * 0.75 * 1.5}px`,
    small: `${baseFontSize * 0.75 * 1.5}px`,
    medium: `${baseFontSize * 1.5}px`,
    large: `${baseFontSize * 1.25 * 1.5}px`,
    xlarge: `${baseFontSize * 1.5 * 1.5}px`,
  };
  return `
    font-size: ${sizes[size]};
    line-height: ${lineHeights[size]};
  `;
};

const StyledButton = styled.button`
  align-items: center;
  display: flex;
  font-weight: bold;
  justify-content: center;
  letter-spacing: 0.2px;
  border-radius: 4px;

  ${props => `cursor: ${props.disabled ? 'not-allowed' : 'pointer'};`}

  ${buttonFontAndLineProps}

  ${({
    size,
    theme: {
      spacing: { large, xlarge, xxlarge, xxxlarge },
    },
  }) => {
    const heights = {
      xsmall: `${large}px`,
      small: `${xlarge}px`,
      medium: `${xxlarge}px`,
      large: `${xxxlarge}px`,
    };

    return `min-height: ${heights[size]};`;
  }}

  ${({
    size,
    theme: {
      spacing: { small, medium, xsmall, xxsmall, large },
    },
  }) => {
    const paddings = {
      xsmall: `${xxsmall}px ${small}px`,
      small: `${small / 2}px ${small}px`,
      medium: `${xsmall}px ${medium}px`,
      large: `${xsmall * 1.25}px ${medium}px`,
      xlarge: `${large / 2}px ${medium}px`,
    };

    return `padding: ${paddings[size]};`;
  }}

  ${props =>
    props.center &&
    `
    margin-left: auto;
    margin-right: auto;
  `}

  transition: all 0.2s ease-in-out;

  ${props => props.full && `width: 100%;`}

  ${({ skin, theme, disabled, stroked }) => {
    const {
      components: {
        button: {
          skins: {
            [skin]: {
              mainColor: {
                100: mainColor100,
                500: mainColor500,
                700: mainColor700,
              },
              text: { 100: text100 },
            },
          },
        },
      },
      colors: {
        neutral: { 100: neutral100, 500: neutral500 },
      },
    } = theme;

    let bgColor;
    let textColor;

    if (disabled && stroked) {
      bgColor = neutral100;
      textColor = neutral500;
    } else if (stroked) {
      bgColor = neutral100;
      textColor = mainColor500;
    } else if (disabled) {
      bgColor = neutral500;
      textColor = neutral100;
    } else {
      bgColor = mainColor500;
      textColor = text100;
    }

    return `
      background-color: ${bgColor};
      color: ${textColor};

      border: 2px solid ${disabled ? neutral500 : mainColor500};

      ${shadow(2, neutral500)({ theme })}

      :hover {
        ${
          !disabled
            ? `
              ${shadow(4, mainColor700)({ theme })}
              background-color: ${stroked ? mainColor100 : mainColor700};
              border-color: ${mainColor700};
            `
            : ''
        }
      }

      :focus,
      :focus-within {
        ${!disabled ? shadow(4, mainColor500)({ theme }) : ''}
      }

      :active {
        ${
          !disabled
            ? `
              ${shadow(8, mainColor700)({ theme })}
              background-color: ${stroked ? mainColor100 : mainColor700};
            `
            : ''
        }
      }

    ${ButtonIcon} {
      ${({
        size,
        theme: {
          spacing: { xxsmall, medium, large },
        },
      }) => {
        const sizes = {
          xsmall: `${medium}px`,
          small: `${medium}px`,
          medium: `${large}px`,
          large: `${large}px`,
          xlarge: `${large}px`,
        };

        return `
          font-size: ${sizes[size]};
          margin-right: ${xxsmall}px;
        `;
      }}
    }
    `;
  }}
`;

const Button = ({ children, icon, size, $as, ...rest }) => (
  <StyledButton as={$as} {...rest} size={size}>
    {icon && <ButtonIcon size={size} name={icon} {...rest} />}
    {children}
  </StyledButton>
);

Button.defaultProps = {
  center: false,
  disabled: false,
  stroked: false,
  full: false,
  icon: '',
  size: 'medium',
  skin: 'primary',
  type: 'button',
  children: undefined,
  $as: undefined,
  onClick: () => {},
  theme: {
    colors: defaultColors,
    baseFontSize: defaultBaseFontSize,
    spacing,
    breakpoints: defaultTheme.breakpoints,
    components: {
      button: components.button,
    },
  },
};

Button.propTypes = {
  center: PropTypes.bool,
  disabled: PropTypes.bool,
  stroked: PropTypes.bool,
  full: PropTypes.bool,
  /** Icon name. The full catalogue can be found
   * [here](/?selectedKind=1.%20Foundation&selectedStory=Icons) */
  icon: PropTypes.string,
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
  skin: PropTypes.oneOf([
    'neutral',
    'primary',
    'secondary',
    'success',
    'warning',
    'error',
  ]),
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
  /** https://www.styled-components.com/docs/api#as-polymorphic-prop */
  $as: PropTypes.oneOfType([PropTypes.element, PropTypes.string]),
  onClick: PropTypes.func,
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
    colors: PropTypes.object,
    spacing: PropTypes.object,
    breakpoints: PropTypes.arrayOf(PropTypes.any),
    components: PropTypes.shape({
      button: PropTypes.object,
    }),
  }),
};

const IconButton = styled(Button)`
  ${({ skin, theme }) => {
    const {
      components: {
        button: {
          skins: {
            [skin]: {
              mainColor: {
                300: mainColor300,
                500: mainColor500,
                700: mainColor700,
              },
            },
          },
        },
      },
    } = theme;

    return `
    border-radius: 50%;
    border: none;
    color: ${hexToRgba(mainColor500, 0.5)};
    width: 40px;

    background-color: transparent;
    box-shadow: none;
    outline: none;

    ${ButtonIcon} {
      margin-right: 0;
    }

    :hover,
    :focus {
      box-shadow: none;
      background-color: ${hexToRgba(mainColor300, 0.4)};
      color: ${mainColor700};
    }

    :active {
      box-shadow: none;
      background-color: ${hexToRgba(mainColor300, 0.5)};
      color: ${mainColor700};
    }
    `;
  }}
`;

IconButton.propTypes = {
  size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large', 'xlarge']),
  skin: PropTypes.oneOf([
    'neutral',
    'primary',
    'secondary',
    'success',
    'warning',
    'error',
  ]),
  theme: PropTypes.shape({
    baseFontSize: PropTypes.number,
    colors: PropTypes.object,
    spacing: PropTypes.object,
    components: PropTypes.shape({
      button: PropTypes.object,
    }),
  }),
};

IconButton.defaultProps = {
  size: 'medium',
  skin: 'neutral',
  theme: {
    breakpoints: defaultTheme.breakpoints,
    gutter: '8px',
    baseFontSize: defaultBaseFontSize,
    spacing,
    components: {
      button: components.button,
    },
  },
};

Button.Icon = IconButton;

export default Button;
