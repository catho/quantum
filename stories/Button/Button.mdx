import { Meta, Canvas, Story, ArgsTable } from '@storybook/addon-docs';
import { Header } from '../shared';
import Button from '../../components/Button';
import { 
  Default, 
  FullWidth, 
  Centered,
  WithSkin,
  Disabled,
  WithSize,
  WithIcon
} from './Button.stories.jsx';

<Meta title="Button" component={Button}  />

<Header title="Button">
    Buttons express the action that will occur when the user clicks
    or touches it. Buttons are used to start an action, either in the
    background or foreground of an experience.
</Header>

- [Importing and usage](#importing)
- [API](#api)
- [Examples](#examples)

## <a id="importing"></a>Importing and usage

```js
import Button from '@catho/quantum/Button';
```

<Canvas>
  <Story id="button--default"/>
</Canvas>

## <a id="api"></a>API

<ArgsTable of={Button} />

## <a id="examples"></a>Examples

### Full width

<Canvas>
  <Story id="button--full-width"/>
</Canvas>

### Centered

<Canvas>
  <Story id="button--centered"/>
</Canvas>

### Disabled

<Canvas>
  <Story id="button--disabled"/>
</Canvas>

### With skin

<Canvas>
  <Story id="button--with-skin"/>
</Canvas>

### With size

<Canvas>
  <Story id="button--with-size"/>
</Canvas>

### With Icon

The full catalogue of icons can be found [here](?path=/docs/foundation-icons--page#catalog)

<Canvas>
  <Story id="button--with-icon"/>
</Canvas>
