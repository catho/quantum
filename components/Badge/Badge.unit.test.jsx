import React from 'react';
import renderer from 'react-test-renderer';

import Badge from './Badge';
import Button from '../Button';

describe('<Badge />', () => {
  describe('Should match snapshot', () => {
    it('With number', () => {
      expect(renderer.create(<Badge number={10} />).toJSON()).toMatchSnapshot();
    });

    it('with children', () => {
      expect(
        renderer
          .create(
            <Badge number={10}>
              <Button />
            </Badge>,
          )
          .toJSON(),
      ).toMatchSnapshot();
    });

    it('with skin', () => {
      expect(
        renderer.create(<Badge number={10} skin="success" />).toJSON(),
      ).toMatchSnapshot();
    });
  });
});
