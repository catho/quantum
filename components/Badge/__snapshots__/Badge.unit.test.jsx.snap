// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> Should match snapshot With number 1`] = `
.c0 {
  display: inline-block;
  margin-left: 8px;
  margin-right: 8px;
}

.c1 {
  border-radius: 8px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  height: 24px;
  line-height: 24px;
  min-width: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  padding-left: 2px;
  padding-right: 2px;
  background-color: #f2f2f2;
  color: #424242;
}

<div
  className="c0"
  value={10}
>
  <span
    className="c1"
    value={10}
  >
    10
  </span>
  
</div>
`;

exports[`<Badge /> Should match snapshot with children 1`] = `
.c0 {
  display: inline-block;
  position: relative;
}

.c1 {
  border-radius: 8px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  height: 24px;
  line-height: 24px;
  min-width: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  padding-left: 2px;
  padding-right: 2px;
  position: absolute;
  right: -10px;
  top: -10px;
  background-color: #f2f2f2;
  color: #424242;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 4px;
  cursor: pointer;
  font-size: 16px;
  line-height: 24px;
  min-height: 40px;
  padding: 4px 16px;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
  background-color: #1250C4;
  color: #ffffff;
  border: 2px solid #1250C4;
  box-shadow: 0px 3px 1px -2px rgba(153,153,153,0.2),0px 2px 2px 0px rgba(153,153,153,0.14),0px 1px 5px 0px rgba(153,153,153,0.12);
}

.c2:hover {
  box-shadow: 0px 2px 4px -1px rgba(0,47,123,0.2),0px 4px 5px 0px rgba(0,47,123,0.14),0px 1px 10px 0px rgba(0,47,123,0.12);
  background-color: #002F7B;
  border-color: #002F7B;
}

.c2:focus,
.c2:focus-within {
  box-shadow: 0px 2px 4px -1px rgba(18,80,196,0.2),0px 4px 5px 0px rgba(18,80,196,0.14),0px 1px 10px 0px rgba(18,80,196,0.12);
}

.c2:active {
  box-shadow: 0px 5px 5px -3px rgba(0,47,123,0.2),0px 8px 10px 1px rgba(0,47,123,0.14),0px 3px 14px 2px rgba(0,47,123,0.12);
  background-color: #002F7B;
}

.c2 .Button__ButtonIcon-sc-1ovnfsw-0 function (_ref6) {
  -webkit-var: size = _ref6.size, _ref6$theme$spacing = _ref6.theme.spacing, xxsmall = _ref6$theme$spacing.xxsmall, medium = _ref6$theme$spacing.medium, large = _ref6$theme$spacing.large;
  -moz-var: size = _ref6.size, _ref6$theme$spacing = _ref6.theme.spacing, xxsmall = _ref6$theme$spacing.xxsmall, medium = _ref6$theme$spacing.medium, large = _ref6$theme$spacing.large;
  -ms-var: size = _ref6.size, _ref6$theme$spacing = _ref6.theme.spacing, xxsmall = _ref6$theme$spacing.xxsmall, medium = _ref6$theme$spacing.medium, large = _ref6$theme$spacing.large;
  var: size = _ref6.size, _ref6$theme$spacing = _ref6.theme.spacing, xxsmall = _ref6$theme$spacing.xxsmall, medium = _ref6$theme$spacing.medium, large = _ref6$theme$spacing.large;
  return: "\\n          font-size: ".concat(fontSizes[size],";\\n          margin-right: ").concat(xxsmall,"px;\\n        ");
}

.c2 .Button__ButtonIcon-sc-1ovnfsw-0 function (_ref6) var fontSizes = {
  xsmall: "".concat(medium,"px"), small:"".concat(medium,"px"), medium:"".concat(large,"px"), large:"".concat(large,"px"), xlarge:"".concat(large,"px");
}

<div
  className="c0"
  value={10}
>
  <span
    className="c1"
    value={10}
  >
    10
  </span>
  <button
    className="c2"
    disabled={false}
    onClick={[Function]}
    size="medium"
    type="button"
  >
    
  </button>
</div>
`;

exports[`<Badge /> Should match snapshot with skin 1`] = `
.c0 {
  display: inline-block;
  margin-left: 8px;
  margin-right: 8px;
}

.c1 {
  border-radius: 8px;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
  height: 24px;
  line-height: 24px;
  min-width: 32px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 12px;
  padding-left: 2px;
  padding-right: 2px;
  background-color: #dcedc8;
  color: #004D40;
}

<div
  className="c0"
  value={10}
>
  <span
    className="c1"
    value={10}
  >
    10
  </span>
  
</div>
`;
